src/lib/hooks/swap/useSwapDefaults.ts

```diff
@@ -0,0 +1,68 @@
+import { DefaultAddress } from 'lib/components/Swap'
+import { useSwapAmount, useSwapCurrency } from 'lib/hooks/swap'
+import useActiveWeb3React from 'lib/hooks/useActiveWeb3React'
+import { useToken } from 'lib/hooks/useCurrency'
+import { Field } from 'lib/state/swap'
+import { useCallback, useEffect, useState } from 'react'
+
+export default function useSwapDefaults(
+  defaultInputAddress: DefaultAddress | undefined,
+  defaultInputAmount: string | undefined,
+  defaultOutputAddress: DefaultAddress | undefined,
+  defaultOutputAmount: string | undefined
+) {
+  const [, updateSwapInputCurrency] = useSwapCurrency(Field.INPUT)
+  const [, updateSwapOutputCurrency] = useSwapCurrency(Field.OUTPUT)
+  const [, updateSwapInputAmount] = useSwapAmount(Field.INPUT)
+  const [, updateSwapOutputAmount] = useSwapAmount(Field.OUTPUT)
+  const { chainId } = useActiveWeb3React()
+
+  const inputAddress =
+    typeof defaultInputAddress === 'string'
```

Comment:
All of this logic should be moved to a helper function in the same file, something like `useDefaultToken(address: DefaultAddress): ReturnType<typeof useToken>`.